django-stubs/db/models/sql/compiler.pyi

```diff
@@ -18,6 +18,10 @@ _ParamT: TypeAlias = str | int
 _ParamsT: TypeAlias = list[_ParamT]
 _AsSqlType: TypeAlias = tuple[str, _ParamsT]
 
+class PositionRef(Ref):
+    def __init__(self, ordinal: str, refs: str, source: Expression) -> None: ...
+    def as_sql(self, compiler: SQLCompiler, connection: BaseDatabaseWrapper) -> _AsSqlType: ...
```

Comment:
`as_sql` exists here: https://github.com/typeddjango/django-stubs/blob/de6d1fa991a747de870815cf1b9be562cb9c05f4/django-stubs/db/models/expressions.pyi#L114

`Ref` is a subclass of `BaseExpression`